pygen/pygen_src/test/riscv_instr_base_test.py

```diff
@@ -22,10 +22,12 @@
 class riscv_instr_base_test:
     def __init__(self):
         pass
-    asm = riscv_asm_program_gen()
+
     for _ in range(cfg.num_of_tests):
         cfg.randomize()
+        asm = riscv_asm_program_gen()
         riscv_instr_ins.create_instr_list(cfg)
-        test_name = "riscv_asm_test_{}.S".format(_)
+        test_name = "riscv_arithmetic_basic_test_{}.S".format(_)
```

Comment:
@hodjat just something to keep in mind - you'll have to update this hardcoding when integrating pygen into the end-to-end flow